### PR TITLE
Have BOT_ID depend on environment

### PR DESCRIPTION
**Pull request by JosephMart**

### CODE REVIEW

File: src/services/EnvService.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  private get getProdBotID(): string {
+    return String(process.env.PROD_BOT_ID);
+  }
+
+  private get getDevBotID(): string {
+    return String(process.env.DEV_BOT_ID);
+  }
+}
```
Comment (tristan957) [inline src/services/EnvService.ts]: I think the getters being named `getX` is redundant. Kind of defeats the purpose of syntactic sugar imo.

**Review by tristan957 — CHANGES_REQUESTED** [review]
Add a `container.register(EnvConfigService, { useValue: new EnvConfigService(groupMeInfo.group_id })` in callbacks. That way we can inject into places.

**Review by tristan957 — APPROVED** [review]

File: src/services/EnvConfigService.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    this.opts = opts;
+    console.log(`EnvConfigServiceOpts: ${JSON.stringify(this.opts)}`);
+  }
+
+  /**
+   * BotID returns Dev BOT_ID by default
+   * If PROD GroupMe ID is passed to CTOR, Prod BOT_ID is returned
+   */
+  public get BotID(): string {
+    return this.opts.groupID !== this.ProdGroupID ? this.ProdBotID : this.DevBotID;
```
Comment (JosephMart) [inline src/services/EnvConfigService.ts]: when ready to merge, this should be `return this.opts.groupID !== this.ProdGroupID ? this.ProdBotID : this.DevBotID;`

**Comment by JosephMart** [comment]
Zeit now has preview and prod env variables so a lot of this work can be abstracted to use that instead :)

File: src/services/EnvConfigService.ts (hunk 2 above)
Comment (tristan957) [reply src/services/EnvConfigService.ts]: fix ye?

File: api/loki.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ import { EnvConfigService } from "../src/services/EnvConfigService";
 export default async (req: NowRequest, res: NowResponse): Promise<void> => {
   const groupMeInfo = req.body as GroupMeInfo;
 
-  container.register(EnvConfigService, { useValue: new EnvConfigService({ groupID: groupMeInfo.group_id }) });
+  container.register(EnvConfigService, { useValue: new EnvConfigService() });
```
Comment (tristan957) [inline api/loki.ts]: I think this line can actually be removed since it is being injected somewhere...Needs to be tested though

**Review by tristan957 — APPROVED** [review]

**Pull request merged**